### PR TITLE
Merge story_fragment.xml into single xml file #3576

### LINKED ISSUE #3576: Merge story_fragment.xml into single xml file
Currently there are 3 versions of `story_fragment.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by shivambh12**
Explanation
Fixes [#3576](https://github.com/oppia/oppia-android/issues/3576)

Essential Checklist
 The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
 Any changes to scripts/assets files have their rationale included in the PR explanation.
 The PR follows the style guide.
 The PR does not contain any unnecessary code changes from Android Studio (reference).
 The PR is made from a branch that's not called "develop" and is up-to-date with "develop".
 The PR is assigned to the appropriate reviewers (reference).

Before Changes
Phone
After Changes
Phone(1)
Before Changes
Phone-land
After Changes
Phone-land(1)
Before Changes
Tablet
After Changes
Tablet(1)
Before Changes
Tablet-land
After Changes
Tablet-land(1)

### CODE REVIEW

**Comment by shivambh12** [comment]
@ARJUPTA  PTAL

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+  <!-- Story Fragment dimens. -->
+  <dimen name="padding_bottom">160dp</dimen>
+  <dimen name="custom_match_parent">-1px</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: @shivambh12 We cannot use negative values. If its not possible to merge all files without this solution then its fine to keep multiple files. Try to merge as many as you can without setting values in `negative`.

Also, please define all you dimens in a meaningful way (check names of all other values) and also they should be defined at the bottom on the xml file.

cc: @ARJUPTA

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL.

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: @rt4914  I will define names of dimens meaningful way and also at bottom of xml.But, I tried to do with 0 dp(match_parent) instead of -1px but it doesn't work. So, I should leave two xml files one for phone and one for tablet.

**Comment by shivambh12** [comment]
@rt4914  PTAL

**Comment by rt4914** [comment]
@ARJUPTA Assigning you first. Thanks.

**Comment by Arjupta** [comment]
I will be able to review this PR by weekend.

File: app/src/main/res/values-land/dimens.xml [hunk 2]
```diff
@@ -384,10 +385,14 @@
   <dimen name="profile_rename_activity_profile_rename_input_margin_end">168dp</dimen>
   <dimen name="profile_rename_activity_profile_rename_save_button_margin_end">168dp</dimen>
 
+  <!-- Story Fragment- land dimens -->
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Nit
`<!-- StoryFragment -->`

Also always define new values at the bottom.

File: app/src/main/res/values-land/dimens.xml [hunk 3]
```diff
@@ -384,10 +385,14 @@
...
+  <dimen name="story_fragment_activity_padding_bottom">60dp</dimen>
+  
   <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">72dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">72dp</dimen>
 
+
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Remove this extra blank line.

File: app/src/main/res/values-land/dimens.xml [hunk 4]
```diff
@@ -384,10 +385,14 @@
   <dimen name="profile_rename_activity_profile_rename_input_margin_end">168dp</dimen>
   <dimen name="profile_rename_activity_profile_rename_save_button_margin_end">168dp</dimen>
 
+  <!-- Story Fragment- land dimens -->
+  <dimen name="story_fragment_activity_padding_bottom">60dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: The name cannot have `activity` as well as `fragment` please use only one based on which file is using this value.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Remove this blank line

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 6]
```diff
@@ -390,6 +391,8 @@
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
 
+  <!-- Story Fragment dimens. -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Nit
`<!-- Story Fragment dimens -->`

Also define new values at bottom.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Remove this blank line

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 8]
```diff
@@ -392,6 +393,8 @@
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
 
+  <!-- Story Fragment dimens. -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Also define new values at the bottom

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Ditto

File: app/src/main/res/values/dimens.xml [hunk 10]
```diff
@@ -582,6 +583,8 @@
   <dimen name="profile_rename_activity_profile_rename_input_margin_end">28dp</dimen>
   <dimen name="profile_rename_activity_profile_rename_save_button_margin_end">28dp</dimen>
 
+  <!-- Story Fragment dimens. -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
@shivambh12 PTAL

**Comment by shivambh12** [comment]
@rt4914  PTAL

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 11]
```diff
@@ -389,11 +389,12 @@
   <dimen name="solution_summary_margin_top_linear_layout">0dp</dimen>
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
-
   <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">36dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">96dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">96dp</dimen>
+  <!-- Story Fragment dimens. -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Just name it as `<!-- StoryFragment -->`

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 12]
```diff
@@ -391,11 +391,12 @@
   <dimen name="solution_summary_margin_top_linear_layout">0dp</dimen>
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
-
  <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">48dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">120dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">120dp</dimen>
+  <!-- Story Fragment dimens. -->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Just name it as `<!-- StoryFragment -->`

File: app/src/main/res/values/dimens.xml [hunk 13]
```diff
@@ -587,5 +587,7 @@
   <dimen name="section_title_text_view_layout_margin_start">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">32dp</dimen>
+  <!-- Story Fragment dimens. -->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Just name it as `<!-- StoryFragment -->`

File: app/src/main/res/values-land/dimens.xml [hunk 14]
```diff
@@ -390,4 +391,7 @@
   <dimen name="section_title_text_view_layout_margin_top">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">72dp</dimen>
 
+  <!-- Story Fragment- land dimens -->
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Just name it as `<!-- StoryFragment -->`

File: app/src/main/res/values-land/dimens.xml (hunk 3 above)
Comment (rt4914) [reply app/src/main/res/values-land/dimens.xml]: @shivambh12 This is pending. Also please make sure you reply to all the comments.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@shivambh12 Left some comments.

Also please make sure you reply to all the comments.

**Comment by rt4914** [comment]
@ARJUPTA Can you please review this PR. Thanks.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 11 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: okk, I will change it.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 12 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: okk, I will change it.

File: app/src/main/res/values/dimens.xml (hunk 13 above)
Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: okk, I will change it.

File: app/src/main/res/values-land/dimens.xml (hunk 14 above)
Comment (shivambh12) [reply app/src/main/res/values-land/dimens.xml]: okk, I will change it.

File: app/src/main/res/values-land/dimens.xml (hunk 3 above)
Comment (shivambh12) [reply app/src/main/res/values-land/dimens.xml]: okk,I will remove it

**Comment by shivambh12** [comment]
@ARJUPTA  PTAL

**Comment by shivambh12** [comment]
@rt4914  PTAL

File: app/src/main/res/values/dimens.xml [hunk 15]
```diff
@@ -587,5 +587,6 @@
   <dimen name="section_title_text_view_layout_margin_start">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">28dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">32dp</dimen>
-
+  <!-- StoryFragment-->
+  <dimen name="story_fragment_padding_bottom">160dp</dimen>
```
Comment (Arjupta) [inline app/src/main/res/values/dimens.xml]: Add a space above the `<!-- StoryFragment-->` comment. Also remove the `bottom_white_space` dimension from this file as it is no more needed now.

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 16]
```diff
@@ -389,11 +389,11 @@
...
-
   <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">36dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">96dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">96dp</dimen>
-
+  <!-- StoryFragment -->
+  <dimen name="story_fragment_padding_bottom">160dp</dimen>
 </resources>
```
Comment (Arjupta) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: As this dimension value is same as in mobile-portrait `dimens.xml` then you can revert this change.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 17]
```diff
@@ -391,11 +391,11 @@
...
-
  <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">48dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">120dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">120dp</dimen>
-
+  <!-- StoryFragment -->
+  <dimen name="story_fragment_padding_bottom">160dp</dimen>
 </resources>
```
Comment (Arjupta) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: As this dimension value is same as in mobile-portrait `dimens.xml` then you can revert this change.

**Review by Arjupta — CHANGES_REQUESTED** [review]
@shivambh12 there are some Nit changes and other than that everythign looks good to me.

File: app/src/main/res/values/dimens.xml (hunk 15 above)
Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: Okk,I will apply these changes

Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: @ARJUPTA On removing bottom_white_space it is showing error.Maybe it is used in any other xml.

**Comment by shivambh12** [comment]
@ARJUPTA PTAL

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 18]
```diff
@@ -389,11 +389,9 @@
...
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
-
   <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">36dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">96dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">96dp</dimen>
-
 </resources>
```
Comment (Arjupta) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Can you reintroduce the spaces as they were earlier.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 19]
```diff
@@ -391,11 +391,9 @@
...
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
-
  <!-- SectionTitleFragment -->
   <dimen name="section_title_divider_view_layout_margin_top">48dp</dimen>
   <dimen name="section_title_text_view_layout_margin_start">120dp</dimen>
   <dimen name="section_title_text_view_layout_margin_top">32dp</dimen>
   <dimen name="section_title_text_view_layout_margin_end">120dp</dimen>
-
 </resources>
```
Comment (Arjupta) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Ditto

File: app/src/main/res/layout-sw600dp/story_fragment.xml [hunk 20]
```diff
@@ -41,7 +42,7 @@
           android:layout_width="wrap_content"
           android:layout_height="wrap_content"
           android:layout_marginEnd="20dp"
-          android:text="@{viewModel.storyNameLiveData}" />
+          android:text="@{viewModel.storyNameLiveData}"/>
       </androidx.appcompat.widget.Toolbar>
```
Comment (Arjupta) [inline app/src/main/res/layout-sw600dp/story_fragment.xml]: You can add the space back here.

**Review by Arjupta — CHANGES_REQUESTED** [review]
@shivambh12 these are some nit changes to be done and other than that you are good to go. Also do reply over the previous reviews so that I know they have been addressed

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 16 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: I, removed this.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 17 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: ditto

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 18 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: Okk,I will add these spaces.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 19 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: Okk.

File: app/src/main/res/layout-sw600dp/story_fragment.xml (hunk 20 above)
Comment (shivambh12) [reply app/src/main/res/layout-sw600dp/story_fragment.xml]: Okk, I wiil add space there.

File: app/src/main/res/values-land/dimens.xml (hunk 2 above)
Comment (shivambh12) [reply app/src/main/res/values-land/dimens.xml]: I resolved it.

File: app/src/main/res/values-land/dimens.xml (hunk 4 above)
Comment (shivambh12) [reply app/src/main/res/values-land/dimens.xml]: okk, I change it.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 5 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: okk, I change it.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 6 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: okk, I change it.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 7 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: I removed it

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 8 above)
Comment (shivambh12) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: Okk.

File: app/src/main/res/values/dimens.xml (hunk 9 above)
Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: Okk

File: app/src/main/res/values/dimens.xml (hunk 10 above)
Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: Okk

**Comment by shivambh12** [comment]
@ARJUPTA PTAL

**Comment by BenHenning** [comment]
Hi. As of today, some main reviewers have taken time off for the next few weeks, so it may take a little while before we can look at this PR. We appreciate your patience while some of our team members recharge. We'll be fully returning on 4 January 2021.

**Review by Arjupta — APPROVED** [review]
Thanks @shivambh12 PR looks good to me.

**Comment by shivambh12** [comment]
@rt4914 PTAL

File: app/src/main/res/values/dimens.xml [hunk 21]
```diff
@@ -590,5 +590,7 @@
 
   <!-- AudioLanguageFragment -->
   <dimen name="audio_language_recycler_view_padding_bottom">132dp</dimen>
-
+   
+  <!-- StoryFragment -->
+  <dimen name="story_fragment_padding_bottom">60dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: This should be `160dp` please check again.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@shivambh12 Thanks. Suggested nit change

File: app/src/main/res/values/dimens.xml (hunk 21 above)
Comment (shivambh12) [reply app/src/main/res/values/dimens.xml]: Yes, I changes it.

**Comment by shivambh12** [comment]
@rt4914 PTAL

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.
